docs/developer_guide.rst

```diff
@@ -7,17 +7,57 @@ testing and style.
 Before submitting a PR
 ----------------------
 
--  All unit and integration tests must pass (please use the docker based tests; see
-   :ref:`docker-sw-testing`).
+-  If you have general questions, feel free to reach out to the faucet-dev mailing list.
+-  If you are new to FAUCET, or are contemplating a major change, it's recommended to
+   open a github issue with the proposed change. This will enable broad understanding of
+   your work including being able to catch any potential snags very early (for example,
+   adding new dependencies). Architectural and approach questions are best
+   settled at this stage before any code is written.
+-  Please send relatively small, tightly scoped PRs (approx 200-300 LOC or less).
+   This makes review and analysis easier and lowers risk, including risk of merge
+   conflicts with other PRs. Larger changes must be refactored into incremental changes.
 -  You must add a test if FAUCET's functionality changes (ie. a new
    feature, or correcting a bug).
+-  All unit and integration tests must pass (please use the docker based tests; see
+   :ref:`docker-sw-testing`). Where hardware is available, please also run the hardware
+   based integration tests also.
+-  You must use TravisCI testing on your fork/github repo, which enables the maintainers
+   to quickly verify that your changes pass all tests in a pristine environment,
+   and also conserves TravisCI resources on the main branch (by minimizing runs on
+   potentially failing test runs which could be caught before opening a PR on the
+   main branch).
```

Comment:
I think I would make this language less strong, maybe something to the effect of:

"-  In order to speed up acceptance of your PR we recommend enabling TravisCI on your own github repo, and linking the test results in the body of the PR. This enables the maintainers to quickly verify that your changes pass all tests in a pristine environment while conserving our TravisCI resources on the main branch (by minimizing runs on potentially failing test runs which could be caught before opening a PR on the main branch)."

Reply:
OK changed to suggested recommendation not requirement.